aten/src/ATen/native/DispatchStub.h

```diff
@@ -70,8 +70,8 @@ struct CAFFE2_API DispatchStub<rT (*)(Args...), T> {
       // they will still compute the same value for cpu_dispatch_ptr.
       if (!cpu_dispatch_ptr.load(std::memory_order_relaxed)) {
         FnPtr tmp_cpu_dispatch_ptr = nullptr;
-        cpu_dispatch_ptr.compare_exchange_weak(
-            tmp_cpu_dispatch_ptr, choose_cpu_impl(), std::memory_order_relaxed);
+        while(!cpu_dispatch_ptr.compare_exchange_weak(
+            tmp_cpu_dispatch_ptr, choose_cpu_impl(), std::memory_order_relaxed));
```

Comment:
This... still doesn't look right. You don't redo the `cpu_dispatch_ptr.load` when you loop, so if you race and `cpu_dispatch_ptr` has quiesced to a non-nullptr by the time you enter this loop, now you have an infinite loop. Much worse!!

Reply:
If my understanding is correct, if race, the `cpu_dispatch_ptr` has been set to true.,then the `compare_exchange_weak` will return `false`. But in the same time the expected value - `tmp_cpu_dispatch_ptr` will be replaced with the contained value as described in the [doc](http://www.cplusplus.com/reference/atomic/atomic/compare_exchange_weak/). So the in next while loop, the `compare_exchange_weak` will return true, thus break the loop.